File: mobile/ios/src/data/models/session.swift
import Foundation

enum SessionSource: Int, Codable {
  case apple = 0
  case facebook = 1
}

protocol Session: Codable {
  @available(iOS, deprecated: 1.0.0)
  var authenticationToken: String? {get set}
  @available(iOS, deprecated: 1.0.0)
  var refreshToken: String? {get set}
  var lastVenue: String? {get set}
  var cachedVenueLocations: [String:String]? {get set}
  var pushToken: String? {get set}
  var hasSkippedOnboarding: Bool? {get set}
  @available(iOS, deprecated: 1.0.0)
  var phoneNumber: String? {get set}
  @available(iOS, deprecated: 1.0.0)
  var authTokenExpiry: Date? {get set}
  @available(iOS, deprecated: 1.0.0)
  var refreshTokenExpiry: Date? {get set}
  var source: SessionSource? {get set}
}

File: mobile/android/app/src/main/java/io/meshtech/koasta/data/Session.kt
package io.meshtech.koasta.data

import java.util.*

interface Session {
  @Deprecated(message = "Use social auth providers instead")
  var authenticationToken: String?
  @Deprecated(message = "Use social auth providers instead")
  var refreshToken: String?
  var lastVenue: String?
  var cachedVenueLocations: Map<String, String>?
  var pushToken: String?
  var hasSkippedOnboarding: Boolean?
  var phoneNumber: String?
  @Deprecated(message = "Use social auth providers instead")
  var authTokenExpiry: Date?
  @Deprecated(message = "Use social auth providers instead")
  var refreshTokenExpiry: Date?
  var source: Int?
}
